hw/ip/prim/doc/prim_lfsr.md

```diff
@@ -18,49 +18,57 @@ width availability in the lookup table (see below).
+StateOutDw   | int    | Width of the LFSR state to be output (`lfsr_q[StateOutDw-1:0]`)
+DefaultSeed  | logic  | Initial state of the LFSR, must be nonzero for XOR and non-all-ones for XNOR forms.
+CustomCoeffs | logic  | Custom polynomial coefficients of length LfsrDw.
+MaxLenSVA    | bit    | Enables maximum length assertions, use only in sim and FPV.
 
 ## Signal Interfaces
 
-Name          | In/Out | Description
---------------|--------|---------------------------------
-en_i          | input  | Lfsr enable
-data_i[InDw]  | input  | Entropy input
-data_o[OutDw] | output | LFSR state output.
+Name                 | In/Out | Description
+---------------------|--------|---------------------------------
+seed_en_i            | input  | External seed input enable
+seed_i[LfsrDw]       | input  | External seed input
+lfsr_en_i            | input  | Lfsr enable
+entropy_i[EntropyDw] | input  | Entropy input
+state_o[StateOutDw]  | output | LFSR state output.
 
 # Theory of Opeations
 
 ```
-           /-----------\
-en_i       |           |
----------->|   lfsr    |
-           |           |
-data_i     |  LfsrDw   |       data_o
-=====/====>| LfsrType  |=======/=======>
- [InDw]    |   InDw    |    [OutDw]
-           |   OutDw   |
-           |   Seed    |
-           |  Custom   |
-           | MaxLenSVA |
-           |           |
-           \-----------/
+             /----------------\
+seed_en_i    |                |
+------------>|      lfsr      |
+seed_i       |                |
+=====/======>|     LfsrDw     |
+ [LfsrDw]    |    LfsrType    |
+lfsr_en_i    |   EntropyDw    |
+------------>|   StateOutDw   |
+entropy_i    |   DefaultSeed  |  state_o
+=====/======>|  CustomCoeffs  |=====/=======>
+ [EntropyDw] |   MaxLenSVA    |  [StateOutDw]
+             |                |
+             \----------------/
 ```
 
 The LFSR module has an enable input and an additional entropy input that is
-XOR'ed into the LFSR state (connect to zero if this feature is unused). The data
-output contains the lower bits of the LFSR state from `OutDw-1` downto `0`. As
-the entropy input may cause the LFSR to jump into its parasitic state
-(all-zero for XOR, all-ones for XNOR), the LFSR state transition function
-contains a lockup protection which re-seeds the state once this condition is
-detected.
+XOR'ed into the LFSR state (connect to zero if this feature is unused). The
+state output contains the lower bits of the LFSR state from `StateOutDw-1`
+downto `0`. As the entropy input may cause the LFSR to jump into its parasitic
+state (all-zero for XOR, all-ones for XNOR), the LFSR state transition function
+contains a lockup protection which re-seeds the state with `DefaultSeed` once
+this condition is detected.
+
+The LFSR contains an external seed input `seed_i` which can be used to load a
+custom seed into the LFSR by asserting `seed_en_i`. This operation takes
+precedence over internal state updates. If the external seed happens to be a
```

Comment:
The parameter list names look much clearer than before. For the lockup sequence, I am assuming this does not occur (or any action to the LFSR), if the lfsr_en_i is active. In the general, the comments looks pretty clear and concise.

Reply:
thanks for giving it a look. 

lockup is only corrected when the lfsr is enabled (`lfsr_en_i = 1'b1`) and the external seed input is disabled (`seed_en_i = 1'b0`). below is the corresponding ternary assignment:

```systemverilog
  assign lfsr_d = (seed_en_i)           ? seed_i          :
                  (lfsr_en_i && lockup) ? DefaultSeed     :
                  (lfsr_en_i)           ? next_lfsr_state :
                                          lfsr_q;
```